src/memray/_memray/macho_shenanigans.cpp

```diff
@@ -296,7 +298,15 @@ patch_symbols_in_shared_object(
             unsigned int section_type = section->flags & SECTION_TYPE;
             LOG(DEBUG) << "Considering section " << i << " (" << section->segname << ":"
                        << section->sectname << ")";
-            if (_dyld_shared_cache_contains_path(image_name) && strcmp(section->segname, SEG_TEXT) == 0)
+                
+#if defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110000
```

Comment:
I don't think I understand what's happening here. I get that the `#include <AvailabilityMacros.h>` should cause `__MAC_OS_X_VERSION_MAX_ALLOWED` to be defined, but won't it always be defined as `101400` (macOS 10.14) on x86-64 since that's what we set as `MACOSX_DEPLOYMENT_TARGET`?

I'm not really following what this is supposed to be guarding against, actually. Is it that `__builtin_available` wasn't available in 10.14?

Reply:
> I'm not really following what this is supposed to be guarding against, actually. Is it that `__builtin_available` wasn't available in 10.14?

If so, I think we could guard against that with `#if __has_builtin(__builtin_available)` instead.